src/stan/lang/ast/fun/var_occurs_vis_def.hpp

```diff
@@ -62,6 +62,14 @@ namespace stan {
       return false;  // no refs persist out of integrate_ode_control() call
     }
 
+    bool var_occurs_vis::operator()(const algebra_solver& e) const {
+      return false;
```

Comment:
why isn't this and the next function checking recursively into y and theta?  isn't this just looking for expressions with variables through which derivatives need to be taken?